fem/fe.cpp

```diff
@@ -1057,7 +1074,27 @@ void VectorFiniteElement::Project_RT(
    }
    else
    {
-      mfem_error("VectorFiniteElement::Project_RT (fe version)");
```

Comment:
It is not easy to follow the Project* functions, with their different cases (based on map type, square_J, etc.). Before this PR, there was a lack of documentation, so this is not a new issue. It would help me understand things for this review, and others in general, if there were some documentation on the transformations being used and the definition of `I` in the different cases. Some comments about the meaning of `d2n` and `nk` would also be helpful (otherwise, one has to look up the constructor for a particular `FiniteElement` to see what they are).

Reply:
I agree with @dylan-copeland about the documentation. As it is mentioned, I too don't think this is something that needs to be addressed in this PR.

Reply:
@dylan-copeland and @psocratis , I added short comments to both the header and source.  Hopefully they'll make the project methods easier to follow.  Let me know if we need more.